### PR TITLE
feat: SEO

### PR DESCRIPTION
**Pull request by smrtrfszm**

### CODE REVIEW

File: apps/frontend/src/app/components/page-not-found/page-not-found.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  styleUrls: ['./page-not-found.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PageNotFoundComponent implements OnInit, OnDestroy {
+  private _noIndexElement: HTMLElement
+
+  constructor(@Inject(DOCUMENT) private _document) {}
+
+  ngOnInit(): void {
+    this._noIndexElement = document.createElement('meta')
```
Comment (TwoDCube) [inline apps/frontend/src/app/components/page-not-found/page-not-found.component.ts:16]:
```suggestion
    this._noIndexElement = this._document.createElement('meta')
```

You shouldn't use the global document, use this._document instead

File: apps/frontend/src/app/modules/home/components/home/home.component.html [hunk 2]
```diff
@@ -1,26 +1,28 @@
 <div class="container">
   <verseghy-featured-post></verseghy-featured-post>
   <verseghy-alert></verseghy-alert>
```
Comment (TwoDCube) [inline apps/frontend/src/app/modules/home/components/home/home.component.html:3]: I think these are "sections"

File: apps/frontend/src/app/modules/shared/components/card/card.component.html [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
-  <p class="description" *ngIf="post.description">{{ post.description }}</p>
-  <div class="bottom">
-    <img *ngIf="post.author.image" class="author-img" [src]="post.author.image" alt="" />
-    <p>{{ post.author.name }}</p>
-    <p>{{ formatDate(post.date) }}</p>
-  </div>
-  <div class="focus-border"></div>
-</div>
+<a class="content" [ngStyle]="{ 'background-color': 'hsl(' + getColor(post.color) + ' 100% 80%)' }" [routerLink]="['/posts', post.id]">
+  <article>
```
Comment (TwoDCube) [inline apps/frontend/src/app/modules/shared/components/card/card.component.html:2]: This is not an article, this is just a card that points the article, the article is in the posts page

File: apps/frontend/src/app/routing.module.ts [hunk 4]
```diff
@@ -39,6 +40,14 @@ const routes: Routes = [
     path: 'information',
     loadChildren: () => import('./modules/information/information.module').then((m) => m.InformationModule),
   },
+  {
+    path: '404',
+    component: PageNotFoundComponent,
+  },
+  {
+    path: '**',
+    component: PageNotFoundComponent,
```
Comment (TwoDCube) [inline apps/frontend/src/app/routing.module.ts:49]: I'd rather `redirectTo: '/404', pathMatch: 'full'`

File: apps/frontend/src/app/modules/shared/components/page-renderer/page-renderer.component.html [hunk 5]
```diff
@@ -1,2 +1,2 @@
 <h1 *ngIf="title">{{ title }}</h1>
-<div class="content" [innerHTML]="content"></div>
+<article class="content" [innerHTML]="content"></article>
```
Comment (TwoDCube) [inline apps/frontend/src/app/modules/shared/components/page-renderer/page-renderer.component.html:2]: This is not necessarily an `article`, it's only an article when it's called from the `page.component` I'd wrap the page renderer there with an `article` element

**Review by TwoDCube — CHANGES_REQUESTED** [review]

**Review by TwoDCube — CHANGES_REQUESTED** [review]
The posts are not redirecting to 404 if not found

File: apps/frontend/src/app/modules/shared/components/card/card.component.html (hunk 3 above)
Comment (smrtrfszm) [reply apps/frontend/src/app/modules/shared/components/card/card.component.html:2]: MDN suggests to use article for "product cards" too, so I think we should keep it. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article

File: apps/frontend/src/app/modules/shared/components/page-renderer/page-renderer.component.html (hunk 5 above)
Comment (smrtrfszm) [reply apps/frontend/src/app/modules/shared/components/page-renderer/page-renderer.component.html:2]: The article tag represents a self-contained composition in a document and not necessarily just an actual article

**Review by TwoDCube — APPROVED** [review]

**Pull request merged**